### ISSUE TITLE
Bulgarian letters are displayed as X.

### ISSUE THREAD

**Issue by sergop1** [camposantogames/firewatch_localization#17]
Bulgarian letters are displayed as X.
Files downloaded from the repository are stored in UTF-8
How can I fix this?
![default](https://user-images.githubusercontent.com/36797019/36632760-5e80ab8e-1993-11e8-991a-c9704bd7d4a0.png)
I am aware with that Cyrillic isn't supported but i was wondering if i can make somehow russians font support it because Bulgarian language is soo similar with Russian only difference is that Russian language has 1 or 2 more letters

**Comment by adorohovich** [comment]
It seems Cyrillic doesn't work at all (even **basic** Cyrillic set). I checked this by using **Russian keyboard** and got exactly the same result as above:
![screenshot 2018-12-20 23 30 55](https://user-images.githubusercontent.com/452327/50309349-952c3100-04af-11e9-8afc-445ab065b266.png)

@hoverbird any suggestion ?

**Comment by ner00** [comment]
Latest versions of Firewatch support Cyrillic since it includes an official Russian translation.
But why it doesn't work for you, I can only speculate... the support for community-sourced translations was added before those latest updates that included Russian and a bunch of other languages - maybe they didn't extend it to the external loaded subtitles. In any case, there have been revisions to the original text which CampoSanto never bothered to update in this Github project. As far as I see it, their commitment stopped when the game dropped from charts and stopped making money.

**Comment by adorohovich** [comment]
Thank you for response 👍 
Yeah, bummer, if it's true. 

Funny thing that I purchased this game recently only because of this project. Have no intention to play, just wanted to contribute to localization.

**Comment by ner00** [comment]
That's really unfortunate, and I would bet you that you won't get ant official response from the developers, they become little cunts after a while - I'm not even kidding at this point - you can see that they don't even bother to respond to the open issues since 2016, that speaks volumes in relation to their commitment.

**Comment by sergop1** [comment]
So you say it's from the version. Well if that's the case I'll test with a new version.

**Comment by ner00** [comment]
> 
> 
> So you say it's from the version. Well if that's the case I'll test with a new version.

That's a possibility. I'm not completely sure, but since earlier versions weren't released with Cyrillic language support, it may be that the version you are testing with was an earlier build.

**Comment by sergop1** [comment]
Again, they are displayed as xxxxxx, but the strangest part is that even the words that I haven't changed are still displayed as xxxxx. Maybe the key: value is changed in some way for the English version? I've tested with v1.09.

**Comment by sergop1** [comment]
Even more strange is the fact that now I have to change to Russian first and then to my Bulgarian to even display them as xxxx. Otherwise, they stay as the last selected language. Like if I had it on french they will stay on french when I test with my language.

https://imgur.com/a/5Vky2rq

**Comment by sergop1** [comment]
Well i found a way for translation with the help of some people from one resource forum :D
![firewatch_dvumkj7h0k](https://user-images.githubusercontent.com/36797019/50497342-3f233480-0a3f-11e9-9c9c-a38b301051d0.jpg)

For more info if you are interested you can DM me or post here and if i see it ill send the tools.

**Comment by ner00** [comment]
> 
> 
> Well i found a way for translation with the help of some people from one resource forum :D
> ![firewatch_dvumkj7h0k](https://user-images.githubusercontent.com/36797019/50497342-3f233480-0a3f-11e9-9c9c-a38b301051d0.jpg)
> 
> For more info if you are interested you can DM me or post here and if i see it ill send the tools.

Great. But why not post the info here in the open?
If someone like you comes later, they'll have access to the information.

**Comment by sergop1** [comment]
Idk if i'm allowed to post it.
https://zenhax.com/viewtopic.php?f=12&t=2116&hilit=firewatch
You can find all in there.

**Comment by jehad13** [comment]
> Idk if i'm allowed to post it. https://zenhax.com/viewtopic.php?f=12&t=2116&hilit=firewatch You can find all in there.

The website is not working.
Could you send me the tool, please?
I would like to modify the font.

**Comment by ner00** [comment]
> The website is not working.

https://www.zenhax.com/viewtopic.php@t=2116.html

@jehad13 You may not be able to do this, even if you get the tool. The reason for this is because the developers updated the game a year later or so and with the new update they introduced multiple new official translation (including Chinese (Simplified), French, German, Japanese, Russian, Spanish) but then they removed the ability for the game to pull custom translation json from AppData.

Are you sure your version still reads custom translations from `C:\Users\%username%\AppData\LocalLow\CampoSanto\Firewatch\Localization` ?

**Comment by jehad13** [comment]
> > The website is not working.
> 
> https://www.zenhax.com/viewtopic.php@t=2116.html
> 
> @jehad13 You may not be able to do this, even if you get the tool. The reason for this is because the developers updated the game a year later or so and with the new update they introduced multiple new official translation (including Chinese (Simplified), French, German, Japanese, Russian, Spanish) but then they removed the ability for the game to pull custom translation json from AppData.
> 
> Are you sure your version still reads custom translations from `C:\Users\%username%\AppData\LocalLow\CampoSanto\Firewatch\Localization` ?

No, it is not possible to create a custom translation from this path, but I can edit the game files and modify the text using specific tools.
The issue I am facing is adding a new font to be able to input my language and translate the game.

**Comment by ner00** [comment]
> No, it is not possible to create a custom translation from this path, but I can edit the game files and modify the text using specific tools.
> The issue I am facing is adding a new font to be able to input my language and translate the game.

Okay, regarding the tools mentioned by *sergop1*, created by *swuforce*, should be [firewatch import-export.zip](https://github.com/user-attachments/files/18277431/firewatch.import-export.zip), compile as follows:

1. Download and install [AutoIt](https://www.autoitscript.com/site/autoit/downloads/).
2. Copy the files from the the zip `Include` folder to your `Autoit3\Include` folder.
3. [Build](https://www.autoitscript.com/autoit3/docs/intro/compiler.htm) the scripts to exe.

**Comment by jehad13** [comment]
> > No, it is not possible to create a custom translation from this path, but I can edit the game files and modify the text using specific tools.
> > The issue I am facing is adding a new font to be able to input my language and translate the game.
> 
> Okay, regarding the tools mentioned by _sergop1_, created by _swuforce_, should be [firewatch import-export.zip](https://github.com/user-attachments/files/18277431/firewatch.import-export.zip), compile as follows:
> 
> 1. Download and install [AutoIt](https://www.autoitscript.com/site/autoit/downloads/).
> 2. Copy the files from the the zip `Include` folder to your `Autoit3\Include` folder.
> 3. [Build](https://www.autoitscript.com/autoit3/docs/intro/compiler.htm) the scripts to exe.

I tried the tool, but unfortunately, it didn’t work.
Do you know of any tool or method to create a new font for the game (textures)?
I know how to extract the font from the game files, but creating the font is the issue!

**Comment by ner00** [comment]
> Do you know of any tool or method to create a new font for the game (textures)? I know how to extract the font from the game files, but creating the font is the issue!

I'm afraid not. At the time when I was involved I didn't need to change fonts, although there were many people whose languages required that, and this is why I was interested too because I was leading a community project for multiple languages, but we never got around to it and eventually were discouraged by Campo Santo with the removal of the custom translation feature.

The example above from sergop1 is a good one, where he was able to do it. Since the screenshot he showed added Cyrillic before the game officially supported it, there's a good chance that some Russian forums mention this. I'm afraid you'll either have to Google more about it or experiment yourself - although I'm sure you already spent a fair amount of time on your own trying to make it work.

**Comment by ner00** [comment]
@jehad13 Maybe you could try downloading the Ukrainian unofficial translation and look how they implemented the changes: https://www.zoneofgames.ru/games/firewatch/files/8782.html

**Comment by jehad13** [comment]
> @jehad13 Maybe you could try downloading the Ukrainian unofficial translation and look how they implemented the changes: https://www.zoneofgames.ru/games/firewatch/files/8782.html

thank you so much 
I will take a look!

**Comment by ner00** [comment]
In case you eventually figure out how to mod translations and font textures, it would be appreciated for all if you could post back your expertise. Thank you.

**Comment by odocha0986** [comment]
Hello, what can I do with exe file after build?
